### PR TITLE
implement UI for Mipmap. TwisterGE proposal

### PR DESCRIPTION
**Pull request by youle31**
TwisterGE proposal.
Test file: http://pasteall.org/blend/index.php?id=45565

I added optionnal arg mipmap in add filter API. Missing API doc: REMOVED after panzergame review

Do we need to add something in versioning? ANSWERED by panzergame

### CODE REVIEW

File: source/gameengine/Rasterizer/RAS_2DFilterData.h [hunk 1]
```diff
@@ -43,6 +43,8 @@ class RAS_2DFilterData
 	std::vector<std::string> propertyNames;
 	/// The KX_GameObject (or something else?) that provides the values for the uniforms named above.
 	CValue *gameObject;
+	/// Enable/Disable mipmap in 2DFilter actuator at conversion
+	int mipmap;
```
Comment (panzergame) [inline source/gameengine/Rasterizer/RAS_2DFilterData.h]: Why do you not use a bool and do an implicit cast at the assignment ?

File: source/gameengine/Ketsji/KX_2DFilterManager.cpp [hunk 2]
```diff
@@ -101,8 +101,9 @@ KX_PYMETHODDEF_DOC(KX_2DFilterManager, addFilter, " addFilter(index, type, fragm
 	int index = 0;
 	int type = 0;
 	const char *frag = "";
+	int mipmap = 0;
 
-	if (!PyArg_ParseTuple(args, "ii|s:addFilter", &index, &type, &frag)) {
+	if (!PyArg_ParseTuple(args, "ii|s:addFilter", &index, &type, &frag, &mipmap)) {
```
Comment (panzergame) [inline source/gameengine/Ketsji/KX_2DFilterManager.cpp]: I doubt for this part. The user have the possibility to set the mipmap just after he created the filter as he sets uniforms.

**Comment by panzergame** [comment]
@DCubix : If we let the mipmapping disabled as default, you don't need to add a version conversion.

File: source/gameengine/Rasterizer/RAS_2DFilterData.h (hunk 1 above)
Comment (youle31) [inline source/gameengine/Rasterizer/RAS_2DFilterData.h]: There is everywhere int before conversion in ras-2dfilter. Due to bool not supported in dna

File: source/gameengine/Ketsji/KX_2DFilterManager.cpp (hunk 2 above)
Comment (youle31) [inline source/gameengine/Ketsji/KX_2DFilterManager.cpp]: Yeah we Can remove it

**Comment by youle31** [comment]
Ok no need for versioning. Thanks for review

File: source/gameengine/Converter/KX_ConvertActuators.cpp [hunk 3]
```diff
@@ -977,7 +977,7 @@ void BL_ConvertActuators(const char* maggiename,
 			}
 
 			tmp = new SCA_2DFilterActuator(gameobj, filtermode,  _2dfilter->flag,
-			                               _2dfilter->float_arg, _2dfilter->int_arg,
+			                               _2dfilter->float_arg, _2dfilter->int_arg, _2dfilter->mipmap,
```
Comment (panzergame) [inline source/gameengine/Converter/KX_ConvertActuators.cpp]: @youle31: if you set the constructor argument type of mipmap to a bool, the int value from _2dfilter->mipmap will be casted to it implicitly, if you compiler complain to an explicit cast: `bool(value)`.

Comment (youle31) [inline source/gameengine/Converter/KX_ConvertActuators.cpp]: It Is an ont no in filter constructor? I can't check. I'm at Montpellier. I'll see tomorrow

Comment (youle31) [inline source/gameengine/Converter/KX_ConvertActuators.cpp]: Int

Comment (youle31) [inline source/gameengine/Converter/KX_ConvertActuators.cpp]: the nsca_2dfilteractuator constructor is:
SCA_2DFilterActuator(
	        class SCA_IObject* gameobj,
	        int type,
	        short flag,
	        float float_arg,
	        int int_arg,
	        int mipmap,
	        RAS_IRasterizer* rasterizer,
			RAS_2DFilterManager *filterManager,
	        SCA_IScene* scene);

File: source/gameengine/Rasterizer/RAS_2DFilterData.h (hunk 1 above)
Comment (youle31) [inline source/gameengine/Rasterizer/RAS_2DFilterData.h]: I do the cast in ras_2dfilter

File: source/gameengine/Converter/KX_ConvertActuators.cpp (hunk 3 above)
Comment (panzergame) [inline source/gameengine/Converter/KX_ConvertActuators.cpp]: What i mean is that you could let the conversion from int to bool only in this file. And so use bool everywhere in SCA_2DFilterActuator and KX_2DFilterData.

Comment (youle31) [inline source/gameengine/Converter/KX_ConvertActuators.cpp]: okok I'll update. Thanks for review!

Comment (youle31) [inline source/gameengine/Converter/KX_ConvertActuators.cpp]: I changed that. You're right. I didn't realized my mistake. Thanks for have pointing it.

**Pull request closed without merging**